backend/src/test/java/com/jcapucho/backend/controllers/LocationControllerTest.java

```diff
@@ -0,0 +1,270 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(LocationController.class)
+@Import(SecurityConfig.class)
+class LocationControllerTest {
+    @Autowired
+    private WebApplicationContext context;
+
+    private MockMvc mvc;
+
+    @MockitoBean
+    private LocationService locationService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeEach
+    void setup() {
+        mvc = MockMvcBuilders
+                .webAppContextSetup(context)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @Test
+    void createLocationNoAuth() throws Exception {
+        final CreateLocation req = new CreateLocation(
+                "Test",
+                30,
+                20
+        );
+
+        mvc.perform(post("/api/v1/locations/")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content(objectMapper.writeValueAsString(req)))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    void createLocationBadRequest() throws Exception {
+        final CreateLocation req = new CreateLocation(
+                "",
+                180,
+                360
+        );
+
+        mvc.perform(post("/api/v1/locations/")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content(objectMapper.writeValueAsString(req)))
+                .andExpect(status().isBadRequest())
+                .andExpect(content().contentType(MediaType.APPLICATION_PROBLEM_JSON_VALUE));
+
+        Mockito.verify(locationService, Mockito.never()).createLocation(Mockito.any());
+    }
+
+    @Test
+    @WithMockUser
+    void createLocationOk() throws Exception {
+        Mockito.when(locationService.createLocation(Mockito.any())).thenAnswer(i -> {
+            final Location l = i.getArgument(0);
+            l.setId(1L);
+            return l;
+        });
+
+        final CreateLocation req = new CreateLocation(
+                "Good",
+                20,
+                30
+        );
+
+        final LocationDTO expected = new LocationDTO(
+                1L,
+                "Good",
+                20,
+                30
+        );
+
+        mvc.perform(post("/api/v1/locations/")
+                        .contentType(MediaType.APPLICATION_JSON_VALUE)
+                        .content(objectMapper.writeValueAsString(req)))
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON_VALUE))
+                .andExpect(content().json(objectMapper.writeValueAsString(expected)));
+
+        Mockito.verify(locationService, Mockito.times(1)).createLocation(Mockito.any());
+    }
+
+    @Test
```

Comment:
```suggestion
  @ParameterizedTest
  @CsvSource(value = {"190:-190:100:-100:19", "170:-170:80:-80:19","-170:170:80:-80:19"}, delimiter = ':')
```

Adding a parameterized test like this allows for the testing of all the conditions in the controller, and shutting up sonar